### ISSUE TITLE
Dynamic resources directive when defined both in a process and the config file cause an error

### ISSUE THREAD

**Issue by jkaessens** [nextflow-io/nextflow#497]
Dear Paolo,
I fail to set per-process resources in my Nextflow scripts. Whenever I use the any of the dynamic resources directives, I get the following error:

```
N E X T F L O W  ~  version 0.25.5
Launching `test.nf` [gloomy_mcclintock] - revision: d35b2809ed
ERROR ~ No signature of method: nextflow.util.MemoryUnit.multiply() is applicable for argument types: (groovy.util.ConfigObject) values: [[:]]
Possible solutions: multiply(java.lang.Number)
```
For example, this happens when I launch the following `test.nf`:
```
process printVars {
    memory '1GB'
    echo true
    cpus 10
    time '1 day'

    input:
    each x from 1,2,3
    """
    echo indx: ${task.index}
    echo proc: ${task.process}
    echo exec: ${task.executor}
    echo cpus: ${task.cpus}
    echo penv: ${task.penv}
    echo time: ${task.time}
    echo memory: ${task.memory}
    """
}
```
with the following configuration `config.conf`:
```
process {
  executor = 'local'
  memory = { 4.GB * task.attempt }
  cpus = { 1 * task.attempt }
  time = { 1.h * task.attempt }
  errorStrategy = { task.exitStatus == 143 ? 'retry' : 'finish' }
  maxRetries = 3
}
```

The cmdline is
```
nextflow -c config.conf run test.nf
```

A few more findings:
* When running the pipeline without the `-c` argument, i.e. `nextflow test.nf`, everything works fine
* When exchanging the "cpus" and the "memory" line in `test.nf`, the error becomes
```
N E X T F L O W  ~  version 0.25.5
Launching `test.nf` [focused_sammet] - revision: 09e6008b45
ERROR ~ No signature of method: nextflow.util.MemoryUnit.multiply() is applicable for argument types: (groovy.util.ConfigObject) values: [[:]]
Possible solutions: multiply(java.lang.Number)
```

Here is [the full nextflow.log](https://gist.github.com/jkaessens/6b0e43e9a1c6215e508b4c4250185665).

Thank you for looking into it.
This is a follow-up to my [Nextflow group topic](https://groups.google.com/forum/#!topic/nextflow/8_No53ae66c).

**Comment by pditommaso** [comment]
It's a bit weird what's happening here. A temporary workaround is to *not* specify `memory`, `cpus`, etc in the process when they are defined in the config file. In that way it should work. Please confirm.

**Comment by jkaessens** [comment]
Yes, I can confirm that.

**Comment by jkaessens** [comment]
I just found out, that this also works the other way round: if I remove the `memory` or `cpus` directive from the config file, it works fine specifying them in the `test.nf` script.

**Comment by pditommaso** [comment]
Included in version `0.26.0-RC2`.
